tests/Unit/PointwiseFunctions/Xcts/Test_AdmMass.cpp

```diff
@@ -0,0 +1,204 @@
+  const std::array<double, 3> center{{0., 0., 0.}};
+  const std::array<double, 3> boost_velocity{{0., 0., boost_speed}};
+  const KerrSchild solution{mass, dimensionless_spin, center, boost_velocity};
+
+  // Set up domain
+  domain::creators::Sphere shell{
+      /* inner_radius */ 2 * mass,
+      /* outer_radius */ distance,
+      /* interior */ domain::creators::Sphere::Excision{},
+      refinement_level,
+      polynomial_order + 1,
+      true,
+      {},
+      {},
+      domain::CoordinateMaps::Distribution::Logarithmic};
+  const auto shell_domain = shell.create_domain();
+  const auto& blocks = shell_domain.blocks();
+  const auto& initial_ref_levels = shell.initial_refinement_levels();
+  const auto element_ids = initial_element_ids(initial_ref_levels);
+  const Mesh<2> face_mesh{polynomial_order + 1, Spectral::Basis::Legendre,
+                          Spectral::Quadrature::GaussLobatto};
+  const Mesh<3> mesh_3d{polynomial_order + 1, Spectral::Basis::Legendre,
+                        Spectral::Quadrature::GaussLobatto};
+
+  // Initialize surface integral
+  double surface_integral = 0;
+
+  // Compute integrals by summing over each element
+  for (const auto& element_id : element_ids) {
+    // Skip elements not at the outer boundary
+    const size_t radial_dimension = 2;
+    const auto radial_segment_id = element_id.segment_id(radial_dimension);
+    if (radial_segment_id.index() !=
+        two_to_the(radial_segment_id.refinement_level()) - 1) {
+      continue;
+    }
+
+    // Get element information
+    const auto& current_block = blocks.at(element_id.block_id());
+    const auto current_element = domain::Initialization::create_initial_element(
+        element_id, current_block, initial_ref_levels);
+    const ElementMap<3, Frame::Inertial> logical_to_inertial_map(
+        element_id, current_block.stationary_map().get_clone());
+
+    // Loop over external boundaries
+    for (auto boundary_direction : current_element.external_boundaries()) {
+      // Skip interfaces not at the outer boundary
+      if (boundary_direction != Direction<3>::upper_zeta()) {
+        continue;
+      }
+
+      // Get 3-dimensional coordinates
+      const auto logical_coords_3d = logical_coordinates(mesh_3d);
+      const auto inertial_coords_3d =
+          logical_to_inertial_map(logical_coords_3d);
+      const auto inv_jacobian_3d =
+          logical_to_inertial_map.inv_jacobian(logical_coords_3d);
+
+      // Get 3-dimensional conformal factor
+      const auto background_fields_3d = solution.variables(
+          inertial_coords_3d,
+          tmpl::list<Xcts::Tags::ConformalFactor<DataVector>>{});
+      const auto& conformal_factor_3d =
+          get<Xcts::Tags::ConformalFactor<DataVector>>(background_fields_3d);
+
+      // Compute conformal factor derivatives and slice them to the interface
+      const auto deriv_conformal_factor_3d =
+          partial_derivative(conformal_factor_3d, mesh_3d, inv_jacobian_3d);
```

Comment:
You can retrieve this from the solution as `::Tags::deriv<Xcts::Tags::ConformalFactorMinusOne<DataVector>, tmpl::size_t<3>, Frame::Inertial>` (can do this directly on the face)